### DISCUSSION TITLE
`No QueryClient set, use QueryClientProvider to set one` after following React instructions

### DISCUSSION THREAD

**Discussion by gino** [trpc/trpc#2703] [Q&A] [answered]
Hi everyone,

I am trying to integrate TRPC into a Vite project) using the [React instructions](https://trpc.io/docs/v9/react#client-side) from the docs.

Unfortunately, I am having type errors with the `trpc.Provider`, for the `queryClient` prop:
```
Type 'QueryClient' is missing the following properties from type 'QueryClient': cancelMutations, executeMutation, defaultQueryObserverOptionsts(2739)
```

And this also results into an error while trying to use trpc: `No QueryClient set, use QueryClientProvider to set one`.

I realized that the errors disappear when I try to import from `react-query` instead of `@tanstack/react-query`. I thought I fixed all the issues by using this approach. But when I tried to integrate [@tanstack/react-query-devtools](https://tanstack.com/query/v4/docs/devtools) using their official docs, I get the same error as before: `No QueryClient set, use QueryClientProvider to set one`.

I feel like I am doing something wrong by changing the imports in the first place. For some reason, I think that I shouldn't since the docs are probably right and also because the DevTools package breaks.

I have a CodeSandbox example with a fresh React project that just followed all the steps from the [docs](https://trpc.io/docs/v9/react#client-side) that also has the typing error:
https://codesandbox.io/s/vibrant-hypatia-9wkrvf?file=/src/App.tsx

Hope someone is able to help me out, I'd appreciate it!

**Comment by jlalmes** [comment, marked as answer]
tRPC v9 only supports `react-query@3`, see docs here: https://react-query-v3.tanstack.com/

If you wish to use `@tanstack/react-query@4` then you must upgrade to tRPC v10.
